public/views/Search.vue

```diff
@@ -1,110 +1,287 @@
+              :dataset="result.dataset"
+              allow-join
+              allow-import
+            >
+            </dataset-preview>
+            <model-preview
+              v-if="result.type === 'model'"
+              :key="result.model.fittedSolutionId"
+              :model="result.model"
+            >
+            </model-preview>
+          </template>
+        </div>
+      </div>
     </section>
   </div>
 </template>
 
 <script lang="ts">
 import _ from "lodash";
 import Vue from "vue";
+import DatasetPreview from "../components/DatasetPreview";
 import FileUploader from "../components/FileUploader";
 import FileUploaderStatus from "../components/FileUploaderStatus";
-import DatasetPreviewCard from "../components/DatasetPreviewCard";
+import ModelPreview from "../components/ModelPreview";
 import SearchBar from "../components/SearchBar";
-import DatasetSearchResults from "../components/DatasetSearchResults";
-import ModelSearchResults from "../components/ModelSearchResults";
 import { Dataset } from "../store/dataset/index";
-import { Model } from "../store/model/index";
-import { createRouteEntry, overlayRouteEntry } from "../util/routes";
-import { getters as routeGetters } from "../store/route/module";
-import { actions as viewActions } from "../store/view/module";
 import {
   getters as datasetGetters,
   actions as datasetActions
 } from "../store/dataset/module";
+import { Model } from "../store/model/index";
 import { getters as modelGetters } from "../store/model/module";
+import { getters as routeGetters } from "../store/route/module";
 import { SEARCH_ROUTE, JOIN_DATASETS_ROUTE } from "../store/route/index";
+import { actions as viewActions } from "../store/view/module";
+import { createRouteEntry, overlayRouteEntry } from "../util/routes";
+import { spinnerHTML } from "../util/spinner";
 
 export default Vue.extend({
   name: "search-view",
 
   components: {
-    SearchBar,
-    DatasetSearchResults,
-    ModelSearchResults,
-    DatasetPreviewCard,
+    DatasetPreview,
     FileUploader,
-    FileUploaderStatus
+    FileUploaderStatus,
+    ModelPreview,
+    SearchBar
   },
 
   data() {
     return {
       isPending: false,
+      sorting: {
+        asc: true,
+        type: "name"
+      },
+      tab: "datasets",
       uploadData: {},
       uploadStatus: ""
     };
   },
 
   computed: {
-    terms(): string {
-      return routeGetters.getRouteTerms(this.$store);
+    filteredDatasets(): Dataset[] {
+      return datasetGetters.getFilteredDatasets(this.$store);
+    },
+
+    filteredDatasetsIds(): Set<String> {
+      const ids = this.filteredDatasets.map(dataset => dataset.id);
+      return new Set(ids);
+    },
+
+    filteredModels(): Model[] {
+      const models = modelGetters.getModels(this.$store);
+
+      // Only display the models using dataset that the search bar has found.
+      return models.filter(model =>
+        this.filteredDatasetsIds.has(model.datasetId)
```

Comment:
Because the search bar says `Datasets search` I use the `filteredDatasets` results to only show models using those datasets.